storage/storage.c

```diff
@@ -838,6 +868,20 @@ static secbool decrypt_dek(const uint8_t *kek, const uint8_t *keiv) {
     return secfalse;
   }
 
+  uint8_t kek[SHA256_DIGEST_LENGTH] = {0};
+  uint8_t keiv[SHA256_DIGEST_LENGTH] = {0};
+  if (get_lock_version() >= 5) {
+    if (sectrue !=
+        derive_kek_unlock(pin, pin_len, storage_salt, ext_salt, kek)) {
```

Comment:
What is the reason for calling derive_kek_unlock from decrypt_dek instead of unlock? In my opinion, it reduces readability.

Reply:
It improves readability by making the function more similar to `set_pin()`. In fact, I meant to move the reading of the storage salt inside of the function too, but forgot to do it. Done in cd6c2c7079c30108b0592a9f36dddc1f9236dbef.

Reply:
So the only reason for you was to improve readability?

Reply:
I think there was also something that had to do with cleaner error handling, but I am not sure exactly what it might have been.